File: src/main/java/ai/devchat/cli/DevChatResponseConsumer.java
package ai.devchat.cli;

import java.util.function.Consumer;

public class DevChatResponseConsumer implements Consumer<String> {

    private final Consumer<DevChatResponse> responseCallback;
    private final DevChatResponse response;

    public DevChatResponseConsumer(Consumer<DevChatResponse> responseCallback) {
        this.responseCallback = responseCallback;
        this.response = new DevChatResponse();
    }

    @Override
    public void accept(String line) {
        response.populateFromLine(line);

        if (response.getMessage() != null) {
            responseCallback.accept(response);
        }
    }
}

File: src/main/kotlin/ai/devchat/cli/DevChatResponseConsumer.kt
package ai.devchat.cli

import java.util.function.Consumer

class DevChatResponseConsumer(private val responseCallback: Consumer<DevChatResponse>) : Consumer<String> {
    private val response: DevChatResponse

    init {
        response = DevChatResponse()
    }

    override fun accept(line: String) {
        response.populateFromLine(line)
        if (response.message != null) {
            responseCallback.accept(response)
        }
    }
}
